deliver/lib/deliver/upload_metadata.rb

```diff
@@ -462,14 +451,18 @@ def retry_if_nil(message, tries: 5, wait_time: 10)
       end
     end
 
+    # Checking if the metadata to update includes App Info
+    def is_updating_app_info(options)
+      LOCALISED_APP_VALUES.keys.each do |key|
+        return true unless options[key].nil?
```

Comment:
What kind of info does `options` contain? By reading your explanation in the PR it seems like this would contain the info that the user is trying to upload/update, but apparently this is just _deliver_'s options passed by the user, which could be these:

<img width="600" alt="image" src="https://github.com/fastlane/fastlane/assets/8419048/5bded8e6-6e4b-479e-9983-49e894d3b262">

But the user can also update those app store values by modifying the text files in their metadata folder. Idk if _deliver_ already takes those properties into consideration, checks if they need to be updated (i.e. compare local metadata files against what's in the app store connect) 🤔 this would need to be checked

Reply:
Yep, it contains the info the user is trying to upload, including the ones from text files. It uploads them in [load_from_filesystem](https://github.com/fastlane/fastlane/blob/ab9fb3249adaddf6c05655891b4d280415dfc84e/deliver/lib/deliver/upload_metadata.rb#L524) which is [called before upload](https://github.com/fastlane/fastlane/blob/ab9fb3249adaddf6c05655891b4d280415dfc84e/deliver/lib/deliver/runner.rb#L143). 
I've also tested it locally. 

It doesn't compare local data against what's in the App Store Connect tho, so I've added this to the function.